src/dos/cdrom_image.cpp

```diff
@@ -70,80 +89,155 @@ int CDROM_Interface_Image::BinaryFile::getLength()
 	return length;
 }
 
-#if defined(C_SDL_SOUND)
+Bit16u CDROM_Interface_Image::BinaryFile::getEndian()
+{
+	// Image files are read into native-endian byte-order
+#if defined(WORDS_BIGENDIAN)
+	return AUDIO_S16MSB;
+#else
+	return AUDIO_S16LSB;
+#endif
```

Comment:
After looking more in detail in this code… So we have this `getEndian` method, with this implementation ifdefed (the other implementation is not ifdefed, but it's just a getter), and the only usage is also ifdefed, and it's used to overwrite a function pointer indicating which method should be used on a different object…

So this getter is part of overengineered function dispatcher. I don't have a clear idea how to simplify this right now, but let's keep this in mind - there is a room for improvement here.

Reply:
Yes; in `cdda-merge-5`, I've done away with needing any preprocessor statements to deal with endianness, and these endian-related code blocks are now half the lines.